### ISSUE TITLE
boutcore enum handling matches comments

### ISSUE THREAD

**Issue by d7919** [boutproject/BOUT-dev#1349]
Partially related to #1314 

It seems the procedure for identifying valid enum names matches any line containing the text "enum". This unfortunately seems to include inside comments (and presumably this would also match variable names).

If we supported the use of regular expressions in the `grep` call that's used we could perhaps change the grep to search for something like `"^ *\benum\b "` to be a bit more specific that the `enum` must be the first full word on a line. This would obviously still fail in a situation like:

~~~~
/* This is a very useful comment describing
    enum called Foo Bar and other things
*/
~~~~

but that should be easy enough to work around if it ever came up.

**Comment by d7919** [comment]
@dschwoerer do you think the use of a regexp is possible (e.g. portable and accurate)?

**Comment by ZedThree** [comment]
Could boutcore be built using SWIG or pybind11? Using one of these tools might side-step the need to manually parse source files.

**Comment by dschwoerer** [comment]
I think at least that part of parsing needs to be rewritten.
I don't know SWIG or pybind11, but will have a look when I have time for a rewrite ...

**Comment by d7919** [comment]
A temporary fix is in #1384 -- this just uses a regex to tighten the requirements slightly. It seems to work locally and on Travis.

**Comment by d7919** [comment]
Closing as fixed through #1384
